components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1548,6 +1551,9 @@ private boolean isAllConfiguredUserStoresJDBC() {
 
     private int getMaxLimitForTotalResults(String domainName) {
 
+        if (StringUtils.isEmpty(domainName)) {
+            domainName = getPrimaryUserStoreDomain();
+        }
```

Comment:
Fix the 1st problem mentioned here https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/609#issuecomment-2644715275

When domain is null, set the primary domain. Then the subsequent logics like `if (isJDBCUSerStore(domainName)` will be evaluated properly.